src/lib.rs

```diff
@@ -885,51 +925,747 @@ impl GooseAttack {
+        self.users = if let Some(u) = self.configuration.users {
+            if self.attack_mode == GooseMode::Worker {
+                return Err(GooseError::InvalidOption {
+                    option: "--users".to_string(),
+                    value: self.users.to_string(),
+                    detail: "The --users option can not be set together with the --worker flag."
+                        .to_string(),
+                });
+            }
+            u
+        // Otherwise, use default if set, but not on Worker.
+        } else if let Some(u) = self.defaults.users {
+            if self.attack_mode == GooseMode::Worker {
+                0
+            } else {
+                u
+            }
+        // Otherwise use number of CPUs.
+        } else {
+            if self.attack_mode != GooseMode::Manager && self.attack_mode != GooseMode::Worker {
+                info!(
+                    "concurrent users defaulted to {} (number of CPUs)",
+                    self.number_of_cpus
+                );
+            }
+            self.number_of_cpus
+        };
+
+        // Be sure a valid users value was set.
+        if self.users == 0 && self.attack_mode != GooseMode::Worker {
+            return Err(GooseError::InvalidOption {
+                option: "--users".to_string(),
+                value: self.users.to_string(),
+                detail: "The --users option must be set to at least 1.".to_string(),
+            });
+        }
+
+        if self.attack_mode != GooseMode::Manager && self.attack_mode != GooseMode::Worker {
+            debug!("users = {}", self.users);
+        }
+
+        Ok(())
+    }
+
+    // Configure maximum run time if specified, otherwise run until canceled.
+    fn set_run_time(&mut self) -> Result<(), GooseError> {
+        // Use --run-time if set, don't allow on Worker.
+        self.run_time = if !self.configuration.run_time.is_empty() {
+            if self.attack_mode == GooseMode::Worker {
+                return Err(GooseError::InvalidOption {
+                    option: "--run-time".to_string(),
+                    value: "true".to_string(),
+                    detail: "The --run-time option can not be set together with the --worker flag."
+                        .to_string(),
+                });
+            }
+            util::parse_timespan(&self.configuration.run_time)
+        // Otherwise, use default if set, but not on Worker.
+        } else if let Some(r) = self.defaults.run_time {
+            if self.attack_mode == GooseMode::Worker {
+                0
+            } else {
+                r
+            }
+        }
+        // Otherwise the test runs until canceled.
+        else {
+            0
+        };
+
+        if self.run_time > 0 {
+            info!("run_time = {}", self.run_time);
+        }
+
+        Ok(())
+    }
+
+    // Configure how quickly to hatch Goose Users.
+    fn set_hatch_rate(&mut self) -> Result<(), GooseError> {
```

Comment:
@LionsAd This is going into `GooseAttack`, but likely could just overload `configuration`

Reply:
I re-removed this from `GooseAttack` and fully manage it with `configuration`.